src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -170,7 +175,7 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
                     allowExtensionAndOptionalParameters: isAsync);
             if (isAsync)
             {
-                rewrittenCondition = RewriteAwaitExpression(forEachSyntax, rewrittenCondition, node.AwaitOpt, node.AwaitOpt.GetResult.ReturnType, used: true);
+                rewrittenCondition = RewriteAwaitExpression(forEachSyntax, rewrittenCondition, node.AwaitOpt!, node.AwaitOpt!.GetResult!.ReturnType, used: true);
```

Comment:
Why is the `GetResult` dereference safe?

Reply:
Because we have a semantically valid await expression.  I'll add an assertion.

---
In reply to: [390502272](https://github.com/dotnet/roslyn/pull/42068#discussion_r390502272) [](ancestors = 390502272)